### PR TITLE
Improve support for web app deployment intermediate

### PR DESCRIPTION
**Pull request by Cosaquee**
closes #64

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/coi-gov-pl/puppet-jboss/pull/77" target="_blank">ReviewNinja

**Comment by therek** [comment]
:+1:

**Comment by frodoslaw** [comment]
:+1:

File: lib/puppet_x/coi/jboss/factsrefresher.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require_relative 'facts'
+
+class Puppet_X::Coi::Jboss::FactsRefresher
+  class << self
+
+    def refresh_facts
+      Puppet_X::Coi::Jboss::Facts::define_fullconfig_fact
```
Comment (cardil) [inline lib/puppet_x/coi/jboss/factsrefresher.rb]: To properly refresh those fact you need to remove previous implementation and value from them. Setting want help :sob: !fix

File: manifests/deploy.pp [hunk 2]
```diff
@@ -40,8 +41,9 @@
     controller   => $controller,
     ctrluser     => $jboss::internal::runtime::node::username,
     ctrlpasswd   => $jboss::internal::runtime::node::password,
+    runtime_name => $runtime_name,
```
Comment (cardil) [inline manifests/deploy.pp]: Validate runtime name parameter to ensure that runtime name ends with `.ear`, `.rar`, `.jar` or `.war`

**Pull request closed without merging**